lib/interpreters/split_clustering.rb

```diff
@@ -575,6 +576,10 @@ def self.build_partial_service_vrp(service_vrp, partial_service_ids, available_v
       points_ids = sub_vrp.services.map{ |s| s.activity.point.id }.uniq.compact
       sub_vrp.rests = sub_vrp.rests.select{ |r| sub_vrp.vehicles.flat_map{ |v| v.rests.map(&:id) }.include? r.id }
       sub_vrp.relations = sub_vrp.relations.select{ |r| r.linked_ids.all? { |id| sub_vrp.services.any? { |s| s.id == id } } }
+      if entity == :vehicle
+        sub_vrp.relations.delete_if{ |r| r.type == :same_vehicle }
+        sub_vrp.services.each{ |s| s.relations.delete_if{ |r| r.type == :same_vehicle } }
+      end
```

Comment:
I dealt with this relation type like this : if no vehicle partition is in current vrp, there should be no same_vehicle relation : 

1. if there is a vehicle partition, then we should respect same_vehicle constraints in this partition. Once vehicle cluster is done then there is no vehicle partition in vrp anymore so we can ignore this relation because it is already respected by default.
2. if no vehicle partition is provided then this should be replaced by same_route relation

I think it is a good way to do this instead of editing a global variable LINKING_RELATIONS according to some parameters.

Reply:
I just check the code and it is `split_balanced_kmeans` who is telling `build_partial_vrp` what the `entity` is; so I am okay. 

Apparently I hadn't check where the `entity` was coming from when I asked my question. Because I though it was the entity coming from the duplicated original vrp.preprocessing_partitions. 

Otherwise, we are on the same page  ;) 

I guess I also imagined it a bit differently -- a logic closer to a place where we handle the `vrp.preprocessing_partitions` separation but I guess this a place as good as any.